### PR TITLE
Added endpoint to reset password

### PR DESCRIPTION
**Pull request by BichraiX**
Added endpoint from the spec : https://spec.matrix.org/v1.11/client-server-api/#post_matrixclientv3accountpassword

Added function to get the access token from the request headers or query to utils package and replaced the associated code in tom-server, identity-service, and client-server.

Added checks for server capabilities to /add, /bind and /delete from spec : https://spec.matrix.org/v1.11/client-server-api/#capabilities-negotiation

### CODE REVIEW

File: packages/matrix-client-server/src/account/password/index.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+        (obj, userId) => {
+          validateParameters(res, schema, obj, clientServer.logger, (obj) => {
+            realMethod(
+              res,
+              clientServer,
+              obj as RequestBody,
+              userId as string
+            ).catch((e) => {
+              // istanbul ignore next
+              console.error('Error while changing password')
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/index.ts:198]: No console, use logger

File: packages/matrix-client-server/src/account/password/index.ts [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      { field: 'token', value: accessToken as string, operator: '!=' }
+    ]
+  )
+  Promise.all([deleteDevicesPromise, deleteTokenPromise])
+    .then(() => {
+      send(res, 200, {})
+    })
+    .catch((e) => {
+      // istanbul ignore next
+      console.error('Error while deleting devices and token')
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/index.ts:85]: No console, use logger

File: packages/matrix-client-server/src/account/password/index.ts [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+              accessToken,
+              deviceId
+            )
+          } else {
+            send(res, 200, {})
+          }
+        })
+        .catch((e) => {
+          // istanbul ignore next
+          console.error('Error while updating password')
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/index.ts:138]: Same

File: packages/matrix-client-server/src/account/password/index.ts [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+          // istanbul ignore next
+          send(res, 500, errMsg('unknown', e), clientServer.logger)
+        })
+    })
+    .catch((e) => {
+      // istanbul ignore next
+      console.error('Error while hashing password')
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/index.ts:145]: Same

File: packages/matrix-client-server/src/account/password/index.ts [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+              realMethod(
+                res,
+                clientServer,
+                obj as RequestBody,
+                userId as string,
+                data.device_id,
+                token
+              ).catch((e) => {
+                // istanbul ignore next
+                console.error('Error while changing password')
```
Comment (guimard) [inline packages/matrix-client-server/src/account/password/index.ts:174]: Same

File: packages/matrix-client-server/src/account/password/index.ts (hunk 1 above)
Comment (BichraiX) [reply packages/matrix-client-server/src/account/password/index.ts]: Fixed, thank you !

**Pull request merged**